app/models/Document.js

```diff
@@ -90,10 +89,7 @@ export default class Document extends BaseModel {
 
   @action
   share = async () => {
-    const res = await client.post("/shares.create", { documentId: this.id });
-    invariant(res && res.data, "Share data should be available");
-    this.shareUrl = res.data.url;
-    return this.shareUrl;
+    return this.store.rootStore.shares.create({ documentId: this.id });
```

Comment:
Store the `share` record in the shares store correctly instead of just the url string on the document